examples/contexts/README.md

```diff
@@ -0,0 +1,61 @@
+# Using Contexts to share configuration & state
+
+A Panel component can be configured with Contexts to share configuration & state in its DOM tree instead of manually passing them into each component as props/attrs.
+
+To configure a component to provide contexts to all components in its DOM tree (including itself), set `defaultContexts` in its `config` object:
+```js
+get config() {
+  return {
+    defaultContexts: {
+      counter: new Counter(),
+      darkMode: true,
+    },
+  };
+}
+```
+
+Components must declare the names of any contexts it intends to use in its `config` object:
+```js
+get config() {
+  return {
+    contexts: [`counter`, `darkMode`],
+  };
+}
+```
+
+Once you've declared the context names, you can retrieve a context object/value by calling `Component.getContext(contextName)` (or `$component.getContext(contextName)` in jade):
+```jade
+-- /* index.jade */
+div Current count is #{$component.getContext(`counter`).getCount()})
+```
+
+Any object or value can be a context, but you can optionally define callback properties you can define that Panel recognizes and invokes during the component lifecycle:
+```js
+class Counter {
+  constructor() {
+    this._count = 0;
+    this._connectedComponents = new Set();
+  }
+  increment() {
+    this._count++;
+    for (const component of this._connectedComponents) {
```

Comment:
Interesting...it seem to me like this is something that practically any user of `bindToComponent` will want to do (keep an internal collection of components that are using it). I wonder if this should become a basic mixin/class offered directly in Panel. Not a blocker for this PR, just a thought for dev ergonomics.

Reply:
Yeah food for thought, should definitely keep this in mind if we start seeing this boilerplate frequently. Won't be addressed in this PR.